### PR TITLE
Initial code dump

### PR DESCRIPTION
**Pull request by kamaradclimber**
this PR is separated in two commits. The first one is a nearly direct application of code generator. The second one contains the real code

### CODE REVIEW

**Comment by kamaradclimber** [comment]
cc @criteo-cookbooks/sre-core

File: Berksfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+source 'https://supermarket.chef.io'
+
+cookbook 'kitchen-ohai' # required for ec2
+cookbook 'weight-test', path: File.join(File.dirname(__FILE__), 'test/cookbooks/weight-test')
+cookbook 'resource-weight', path: File.dirname(__FILE__)
```
Comment (Annih) [inline Berksfile]: Please use `metadata`  instead

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+source 'https://rubygems.org'
+
+gem 'berkshelf'
+gem 'kitchen-vagrant'
+gem 'chefspec'
+gem 'rake'
+gem 'foodcritic'
+gem 'chef-zero-scheduled-task'
```
Comment (Annih) [inline Gemfile]: You don't need this gem

File: metadata.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name             'resource-weight'
+maintainer       'Criteo SREs'
+maintainer_email 'sre-core@criteo.com'
+license          'Apache License'
+description      'Allow to set weight on resources'
+long_description 'Installs/Configures resource-weight'
+issues_url       'https://github.com/criteo-cookbooks/resource-weight'
+source_url       'https://github.com/criteo-cookbooks/resource-weight'
+version          '0.1.0'
```
Comment (Annih) [inline metadata.rb:9]: OPT: add chef_version support

File: spec/unit/recipes/default_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      )
+      runner.converge(described_recipe)
+    end
+
+    it 'converges successfully' do
+      expect { chef_run }.to_not raise_error
+    end
+
+    describe 'default weight modification by attribute' do
+      it 'set default weight on reboot resource' do
```
Comment (Annih) [inline spec/unit/recipes/default_spec.rb]: TYPO: sets

**Review by Annih — CHANGES_REQUESTED** [review]
LGTM except 4 comments

File: Berksfile (hunk 1 above)
Comment (kamaradclimber) [reply Berksfile]: done

File: Gemfile (hunk 2 above)
Comment (kamaradclimber) [reply Gemfile]: done

File: metadata.rb (hunk 3 above)
Comment (kamaradclimber) [reply metadata.rb:9]: not really made to work in open source but we could try that at some point

File: spec/unit/recipes/default_spec.rb (hunk 4 above)
Comment (kamaradclimber) [reply spec/unit/recipes/default_spec.rb]: done

**Review by Annih — APPROVED** [review]

**Pull request merged**